### PR TITLE
Adds a new `<Picture>` component to the image integration

### PR DESCRIPTION
**Pull request by tony-sull**
## Changes

- adds a new `<Picture />` component for generative responsive `<picture>` elements with multiple source sizes and image formats
- moves logic for automatically calculating `` dimensions into the `getImage()` API for easier reuse
- uses JSX type definitions for `<Image>` and `<Picture>` components

## Testing

- tests added for the new `<Picture />` component
- adds tests for inlined `import()`s
- existing tests cleaned up a bit

## Docs

README updated

### CODE REVIEW

File: packages/integrations/image/src/types.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
-export type { Image } from '../components/index';
-export * from './index';
+/// <reference path="../types/astro-jsx.d.ts" />
```
Comment (Princesseuh) [inline packages/integrations/image/src/types.ts:1]: Is there no way to refer to the `astro-jsx.d.ts` inside Astro instead of copying it here?

**Review by matthewp — APPROVED** [review]
Love all the tests!

File: packages/integrations/image/src/types.ts (hunk 1 above)
Comment (tony-sull) [reply packages/integrations/image/src/types.ts:1]: I couldn't find one 🤷

Not sure if it was just the monorepo striking again, but I had assumed `path="node_modules/astro/astro-jsx.d.ts"` would have done the trick but it didn't resolve

File: packages/integrations/image/components/Image.astro [hunk 2]
```diff
@@ -17,109 +17,15 @@ export interface RemoteImageProps extends TransformOptions, ImageAttributes {
...
 const props = Astro.props as Props;
 
-const imageProps = await resolveProps(props);
-
-const attrs = await getImage(loader, imageProps);
+const attrs = await getImage(loader, props);
 ---
 
-<img {...attrs} />
+<img {...attrs} loading="lazy" decoding="async" />
```
Comment (natemoo-re) [inline packages/integrations/image/components/Image.astro:25]: Love that we're defaulting to `loading="lazy"` but should we allow [`loading="eager"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-loading) to override this? It is occasionally useful for critical images above the fold.

File: packages/integrations/image/components/Picture.astro [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+const { src, sizes, widths, aspectRatio, formats = ['avif', 'webp'], ...attrs } = Astro.props as Props;
+
+const { image, sources } = await getPicture({ loader, src, widths, formats, aspectRatio });
+---
+
+<picture {...attrs}>
+	{sources.map(attrs => (
+	<source {...attrs} {sizes}>))}
+	<img {...image} loading="lazy" decoding="async" />
```
Comment (natemoo-re) [inline packages/integrations/image/components/Picture.astro:32]: Same comment as above.

File: packages/integrations/image/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const PKG_NAME = '@astrojs/image';
+export const ROUTE_PATTERN = '/_image';
+export const OUTPUT_DIR = '/_image';
```
Comment (natemoo-re) [inline packages/integrations/image/src/config.ts:3]: Such a small nit but I tend to call this file `constants.ts`. `config` is a bit overloaded for our use cases.

File: packages/integrations/image/src/get-image.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	let aspectRatio: number;
+
+	// parse aspect ratio strings, if required (ex: "16:9")
+	if (typeof transform.aspectRatio === 'number') {
+		aspectRatio = transform.aspectRatio;
+	} else {
+		const [width, height] = transform.aspectRatio.split(':');
+		aspectRatio = parseInt(width) / parseInt(height);
```
Comment (natemoo-re) [inline packages/integrations/image/src/get-image.ts:31]: I know these globals aren't going anywhere, but I always try to use

```suggestion
		aspectRatio = Number.parseInt(width) / Number.parseInt(height);
```

File: packages/integrations/image/src/types.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/integrations/image/src/types.ts:1]: This doesn't work? It probably won't in the monorepo, but should outside of it.

```
/// <reference path="astro/astro-jsx.d.ts" />
```

**Review by natemoo-re — COMMENTED** [review]
Amazing update, great tests! Just had a few suggestions.

File: packages/integrations/image/components/Image.astro (hunk 2 above)
Comment (tony-sull) [reply packages/integrations/image/components/Image.astro:25]: Yep that's a great idea!  I'll just default it to `lazy` in the user-provided props and let them override it

File: packages/integrations/image/components/Picture.astro (hunk 3 above)
Comment (tony-sull) [reply packages/integrations/image/components/Picture.astro:32]: Yep, on it

File: packages/integrations/image/src/config.ts (hunk 4 above)
Comment (tony-sull) [reply packages/integrations/image/src/config.ts:3]: Yep great catch, I had named it `config` planning on using it to also normalize integration config and never circled back to it

File: packages/integrations/image/src/types.ts (hunk 1 above)
Comment (tony-sull) [reply packages/integrations/image/src/types.ts]: Ah ha!  I didn't realize `<reference type="astro/astro-jsx">` was a separate syntax, that resolves it similar to `import` 🎉

**Review by natemoo-re — APPROVED** [review]
LGTM, very excited for this!

**Pull request merged**